client-v2/src/app/store/user/user-account.effects.ts

```diff
@@ -0,0 +1,136 @@
+import { Injectable } from '@angular/core'
+import { HotToastService } from '@ngneat/hot-toast'
+import { Actions, createEffect, ofType } from '@ngrx/effects'
+import { Store } from '@ngrx/store'
+import { map, mergeMap, of } from 'rxjs'
+import { UserService } from 'src/app/services/user.service'
+import { AppState } from '..'
+import { appActions } from '../app.actions'
+import { userActions } from './user.actions'
+import { UserState } from './user.model'
+
+@Injectable()
+export class UserAccountEffects {
+    constructor(
+        private actions$: Actions,
+        private userService: UserService,
+        private toast: HotToastService,
+        private store: Store<AppState>
+    ) {
+        this.store.subscribe(state => (this.userState = state.user))
+    }
+
+    userState: UserState | null = null
+
+    updateUsername = createEffect(() => {
+        return this.actions$.pipe(
+            ofType(userActions.updateUsername),
+            mergeMap(dto => {
+                const res$ = this.userService.updateUsername(dto)
+
+                return res$.pipe(
+                    this.toast.observe({
+                        loading: 'Updating username...',
+                        success: res => res.successMessage,
+                        error: err => err.error.message,
```

Comment:
We must use the same replacement mechanism for `field-prefixed:Message` to hide the field prefix in all toasts.